core/src/main/java/org/apache/iceberg/rest/http/RequestResponseSerializers.java

```diff
@@ -0,0 +1,162 @@
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.http;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.IOException;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.PartitionSpecParser;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.SortOrderParser;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+
+
+public class RequestResponseSerializers {
+
+  private RequestResponseSerializers() {
+  }
+
+  public static void registerAll(ObjectMapper mapper) {
+    SimpleModule module = new SimpleModule();
+    module
+        .addSerializer(TableIdentifier.class, new RequestResponseSerializers.TableIdentifierSerializer())
+        .addDeserializer(TableIdentifier.class, new RequestResponseSerializers.TableIdentifierDeserializer())
+        .addSerializer(Namespace.class, new RequestResponseSerializers.NamespaceSerializer())
+        .addDeserializer(Namespace.class, new RequestResponseSerializers.NamespaceDeserializer())
+        .addSerializer(Schema.class, new RequestResponseSerializers.SchemaSerializer())
+        .addDeserializer(Schema.class, new RequestResponseSerializers.SchemaDeserializer())
+        .addSerializer(PartitionSpec.class, new RequestResponseSerializers.PartitionSpecSerializer())
+        .addDeserializer(PartitionSpec.class, new RequestResponseSerializers.PartitionSpecDeserializer())
+        .addSerializer(SortOrder.class, new RequestResponseSerializers.SortOrderSerializer())
+        .addDeserializer(SortOrder.class, new RequestResponseSerializers.SortOrderDeserializer())
+        .addSerializer(TableMetadata.class, new RequestResponseSerializers.TableMetadataSerializer());
+    mapper.registerModule(module);
+  }
+
+  public static class NamespaceDeserializer extends JsonDeserializer<Namespace> {
+    @Override
+    public Namespace deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return Namespace.of(jsonNode.asText().split("\\."));
+    }
+  }
+
+  public static class NamespaceSerializer extends JsonSerializer<Namespace> {
+    @Override
+    public void serialize(Namespace namespace, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+      gen.writeString(namespace.toString());
```

Comment:
I think this should probably be a list of strings to avoid losing the hierarchy when `.` is in one of the namespace identifiers.

Reply:
Yeah agreed. I don't think it can be a path parameter that way though. Should we make it a query parameter?